src/app/AttributePathParams.h

```diff
@@ -81,10 +83,49 @@ struct AttributePathParams
         return true;
     }
 
-    ClusterId mClusterId     = kInvalidClusterId;   // uint32
-    AttributeId mAttributeId = kInvalidAttributeId; // uint32
-    EndpointId mEndpointId   = kInvalidEndpointId;  // uint16
-    ListIndex mListIndex     = kInvalidListIndex;   // uint16
+    bool IsAttributePathIntersect(const DataVersionFilter & other) const
+    {
+        VerifyOrReturnError(HasWildcardEndpointId() || mEndpointId == other.mEndpointId, false);
+        VerifyOrReturnError(HasWildcardClusterId() || mClusterId == other.mClusterId, false);
+
+        return true;
+    }
+
+    bool IsAttributePathIntersect(const ConcreteClusterPath & other) const
+    {
+        VerifyOrReturnError(HasWildcardEndpointId() || mEndpointId == other.mEndpointId, false);
+        VerifyOrReturnError(HasWildcardClusterId() || mClusterId == other.mClusterId, false);
+
+        return true;
+    }
+
+    bool IsWildcardRequest(const AttributePathParams & aOther) const
```

Comment:
This function does not make sense to me.  When would we want to use this outside the one place where it's used now, where it never has a wildcard endpoint/cluster?  And should it really return true if endpoint is wildcard but cluster and attribute are not?

What I think we want instead is a function called `IncludesAllAttributesInCluster` which takes as `ConcreteClusterPath` and returns something like `IncludesAttributesInCluster(other) && HasWildcardAttribute()` or so.  And at the callsite of this thing do `if (path.IncludesAllAttributesInCluster(aPath))`.

Reply:
Put differently, this function isn't 'core' to the behavior of this class. It's specific to the algorithm used in the attribute cache and as such, it doesn't belong here since it's not broadly useful to all consumers of this class.

Reply:
yep, agree, updated, thanks